packages/@aws-cdk/aws-pipes-alpha/lib/inputTransformation.ts

```diff
@@ -0,0 +1,115 @@
+import { DefaultTokenResolver, IResolveContext, Lazy, StringConcat, Token, Tokenization } from 'aws-cdk-lib';
+import { IPipe } from './pipe';
+import { DynamicInput, PipeVariable } from './pipeVariable';
+import { unquote } from './unquote';
+
+type StaticString = string;
+type KeyValue = Record<string, string | PipeVariable>;
+type StaticJsonFlat = Record<string, StaticString | KeyValue>;
+type InputTransformJson = Record<string, StaticString | KeyValue | StaticJsonFlat | IInputTransformation[] | StaticString [] | DynamicInput>;
+
+type InputTransformationValue = StaticString | InputTransformJson;
+
+/**
+ * The inputTemplate that is used to transform the input event payload with unquoted variables
+ */
+export interface InputTransformationConfig {
+  /**
+   * The inputTemplate that is used to transform the input event payload
+   */
+  readonly inputTemplate: string;
+}
+
+/**
+ * Transform or replace the input event payload
+ */
+export interface IInputTransformation {
+  /**
+   * Bind the input transformation to the pipe and returns the inputTemplate string.
+   */
+  bind(pipe: IPipe): InputTransformationConfig;
+}
+
+/**
+ * Transform or replace the input event payload
+ */
+export class InputTransformation implements IInputTransformation {
+
+  /**
+   * Creates an InputTransformation from a string.
+   */
+  static fromText(inputTemplate: StaticString): InputTransformation {
+    return new InputTransformation(inputTemplate);
+  }
+
+  /**
+   * Creates an InputTransformation from a jsonPath expression of the input event.
+   */
+  static fromEventPath(jsonPathExpression: string): InputTransformation {
+    if (!jsonPathExpression.startsWith('$.')) {
+      throw new Error('jsonPathExpression start with "$."');
+    }
+    const jsonPath = `<${jsonPathExpression}>`;
+    return new InputTransformation(jsonPath);
+  }
+
+  /**
+   * Creates an InputTransformation from a pipe variable.
+   */
+  static fromObject(inputTemplate: InputTransformJson): InputTransformation {
+    return new InputTransformation(inputTemplate);
+  }
+
+  /**
+   * The inputTemplate that is used to transform the input event payload
+   */
+  private inputTemplate: InputTransformationValue;
+
+  private constructor(inputTemplate: InputTransformationValue) {
+
+    this.inputTemplate = inputTemplate;
+  }
+
+  public bind(pipe: IPipe): InputTransformationConfig {
+    if (typeof this.inputTemplate === 'string') {
```

Comment:
We need to use `!Token.isUnresolved(this.inputTemplate)` instead of `typeof`

Reply:
This case actually has nothing todo with unresolved values. I added a testcase to make sure tokens are handled correctly. This check tests if the input template is already a string value and should be left as is. 
But I get that this check is not self-explanatory. Therefore I introduced another internal type variable so that we now if we need todo check for unquoting or not. In case of a plain string value there is no need for an unquote.

Reply:
Unresolved tokens can be strings though.

Reply:
Yes, that's better.